test/unit/rules/no-curly-component-invocation-test.js

```diff
@@ -6,71 +6,177 @@ function generateError(name) {
   return `You are using the component {{${name}}} with curly component syntax. You should use <${angleBracketName}> instead. If it is actually a helper you must manually add it to the 'no-curly-component-invocation' rule configuration, e.g. \`'no-curly-component-invocation': { allow: ['${name}'] }\`.`;
 }
 
+const SHARED_GOOD = [
+  '{{foo}}',
```

Comment:
This should not be a "good" example. The octane configuration _already_ requires the template to avoid ambiguous paths making `{{foo}}` reserved for an invocation (either a helper or a component). When `foo` is not a local, it should print an error message that is a little different than the more obvious cases (e.g. it should mention that the path is ambiguous and they should either refactor to `{{this.foo}}` or `<Foo />`).

Reply:
This means we assume `{{foo}}` is a component invocation, not a helper invocation?

IMHO that should be configurable then. I've seen enough cases where apps would have been able to use the `no-curly-component-invocation` rule, but haven't migrated to `no-implicit-this` yet 🤔

Reply:
also, see https://github.com/ember-template-lint/ember-template-lint/issues/1026 😉

Reply:
Yes, I have seen #1026 and replied in detail there.

Reply:
> IMHO that should be configurable then. 

Yep, sounds totally reasonable.

Reply:
We can then make it default to making no assumptions about `{{foo}}` _if_ `no-implicit-this` is disabled.

Reply:
done!